L87-core-remove-grpc-insecure-channel-creation-api

```diff
@@ -0,0 +1,126 @@
+    vtable must be maintained by the caller until grpc_channel_destroy
+    terminates. See grpc_channel_args definition for more on this. */
+GRPCAPI grpc_channel* grpc_channel_create(const char* target,
+                                          grpc_channel_credentials* creds,
+                                          const grpc_channel_args* args);
+```
+
+Third, rename `grpc_server_add_secure_http2_port` to `grpc_server_add_http2_port` and move it from `grpc_security.h` to `grpc.h`.
+
+```
+/** Add a HTTP2 over an encrypted link over tcp listener.
+   Returns bound port number on success, 0 on failure.
+   REQUIRES: server not started */
+GRPCAPI int grpc_server_add_http2_port(grpc_server* server, const char* addr,
+                                       grpc_server_credentials* creds);
+```
+
+Fourth, add `grpc_channel_create_from_fd` and `grpc_server_add_channel_from_fd` to `grpc.h`. Note that the current APIs only support
+either insecure channel credentials (at client-side) or insecure server credentials (at server-side), and using other types of
+credentials will result in a failure. In future, the API's will be improved to support other types of credentials.
+
+
+```
+/** Create a secure channel to 'target' using file descriptor 'fd' and passed-in
+    credentials. The 'target' argument will be used to indicate the name for
+    this channel. Note that this API currently only supports insecure channel
+    credentials. Using other types of credentials will result in a failure. */
+GRPCAPI grpc_channel* grpc_channel_create_from_fd(
+    const char* target, int fd, grpc_channel_credentials* creds,
+    const grpc_channel_args* args);
+
+/** Add the connected secure communication channel based on file descriptor 'fd'
+   to the 'server' and server credentials 'creds'. The 'fd' must be an open file
+   descriptor corresponding to a connected socket. Events from the file
+   descriptor may come on any of the server completion queues (i.e completion
+   queues registered via the grpc_server_register_completion_queue API).
+   The 'reserved' pointer MUST be NULL.
+   TODO(hork): add channel_args to this API to allow endpoints and transports
+   created in this function to participate in the resource quota feature.
+   Note that this API currently only supports inseure server credentials
+   Using other types of credentials will result in a failure. */
+GRPCAPI void grpc_server_add_channel_from_fd(grpc_server* server,
+                                             void* reserved, int fd,
+                                             grpc_server_credentials* creds);
+
+```
+Fifth, remove `grpc_insecure_channel_create` and `grpc_server_add_insecure_http2_port` from `grpc.h`. After this change takes effect, the only way of creating gRPC insecure channels will be first creating insecure channel or server credentials, and then invoking `grpc_channel_create` or `grpc_server_add_http2_port` at client or server side, respectively.
```

Comment:
It might be useful to note that this will have negligible or zero impact because this is a core API and not an end user API so users are likely to be affected by this removal (assuming true).

Reply:
i strongly agree with @sanjaypujare : when I first saw the thread on the mailing-list, I almost made my pants dirty and calmed down only after I saw his above comment.

Reply:
Thanks! I updated the text to reflect it.

Reply:
The new text does acknowledge impact on users who "directly interact with Core surface APIs (e.g., by calling insecure channel creation API's)". Assuming such users exist, won't they benefit from deprecation first and then removal of the APIs?

Reply:
I do not think we need to keep backward-compatibility for C-core API users. Here is a quote from @markdroth  - "The C-core API is not a public API, and we don't offer a backward-compatibility guarantee for it. We are required to publish a gRFC for C-core API changes, but we are not required to make non-breaking changes"

Reply:
That's correct, we do not promise backward compatibility for the C-core API.